### PR TITLE
Feature: Increase Test Coverage

### PR DESCRIPTION
**Pull request by vaeng**

### CODE REVIEW

**Comment by vaeng** [comment]
Maybe we should increase the minor version, because some bugs have been removed with this PR

File: tests/testDeviceAccessLib.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        self.assertFalse(new_value_available)
+
+    def testReadNonBlocking(self):
+        # pay attention to use a normal dummy, shared memory dummy will have a race condition that will fail the test
+        some_value = 445
+
+        # discard first, non-blocking read
+        self.async_scalar_push.read()
+
+        # put one in the queue
```
Comment (mhier) [inline tests/testDeviceAccessLib.py:137]: To discriminate from readLatest, you should put two into the queue and make sure that readNonBlocking will read both of them (and not just the second)

File: CMakeLists.txt [hunk 2]
```diff
@@ -60,10 +62,23 @@ find_package(Boost REQUIRED COMPONENTS ${boost_python} ${boost_numpy})
...
+message(STATUS "Numpy header files in: ${numpy_include_dirs}")
+
 # ==============================================================================#
 INCLUDE(${CMAKE_SOURCE_DIR}/cmake/set_default_build_to_release.cmake)
 
-INCLUDE(${CMAKE_SOURCE_DIR}/cmake/set_default_flags.cmake)
+
+# ignore BOOST deprecated headers
+#add_definitions("-DBOOST_ALLOW_DEPRECATED_HEADERS")
+#add_definitions("-DBOOST_BIND_GLOBAL_PLACEHOLDERS")
```
[the remark this answers is not in the document]
Comment (vaeng) [reply CMakeLists.txt:81]: We had a discussion about those and I think the conclusion was no transfer to the template. We should talk about that again.

File: deviceaccessPython.cc [hunk 3]
```diff
@@ -46,6 +46,7 @@
       .def("getUnit", mtca4upy::GeneralRegisterAccessor::getUnit<ChimeraTK::accessorType<userType>>)                   \
       .def("getName", mtca4upy::GeneralRegisterAccessor::getName<ChimeraTK::accessorType<userType>>)                   \
       .def("getId", mtca4upy::GeneralRegisterAccessor::getId<ChimeraTK::accessorType<userType>>)                       \
+      .def("getAccessModeFlagsString", mtca4upy::GeneralRegisterAccessor::getAccessModeFlagsString<ChimeraTK::accessorType<userType>>)     \
```
[the remark this answers is not in the document]
Comment (vaeng) [reply deviceaccessPython.cc:49]: It is a fix. The function was already there, but it was not working correctly.

File: tests/testDeviceAccessLib.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+import sys
+import unittest
+import numpy as np
+import os
+import threading
+from time import sleep
+
+
+# fmt: off
+# This is a hack for nw. What this does is, add the build directory to python's
```
[the remark this answers is not in the document]
Comment (vaeng) [reply tests/testDeviceAccessLib.py:12]: "now"

File: tests/testDeviceAccessLib.py [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        self.assertTrue(self.rw_int32_acc.isWriteable())
+        self.assertFalse(self.ro_int32_acc.isWriteable())
+        self.assertTrue(self.wo_int32_acc.isWriteable())
+
+        self.assertFalse(self.rw_int32_acc.isReadOnly())
+        self.assertTrue(self.ro_int32_acc.isReadOnly())
+        self.assertFalse(self.wo_int32_acc.isReadOnly())
+
+    def testReadLatest(self):
+        # pay attention to use a normal dummy, shared memory dummy will have a race condition that will fail the test
```
[the remark this answers is not in the document]
Comment (vaeng) [reply tests/testDeviceAccessLib.py:109]: The shared dummy was used to test if everything was working as intended with the help of qthardmon.
The race condition was discussed shortly with @killenb in the daily.

File: tests/testDeviceAccessLib.py (hunk 1 above)
Comment (vaeng) [reply tests/testDeviceAccessLib.py:137]: Good catch! I will add that.

File: CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (mhier) [reply CMakeLists.txt:81]: for sure it should not stay there commented...

**Pull request merged**